curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-telephone-number-validator-project/build-a-telephone-number-validator.md

```diff
@@ -510,6 +511,75 @@ checkBtn.click();
 assert.strictEqual(resultsDiv.innerText.trim().toLowerCase(), 'invalid us number: 11 555-555-5555');
 ```
 
+When the `#user-input` element contains a valid US number and the `#check-btn` element is clicked, the `#results-div` element should contain the text `"Valid US number: "` followed by the number.
+
+```js
+let passedTests = 0;
+
+function generatePhoneNumber(type) {
+  let result = "";
+  let bit1 = "";
+  let bit2 = "";
+  let bit3 = "";
+  
+  for (let i = 0; i < 3; i++) {
+    bit1 += String(Math.floor(Math.random() * 9));
+  }
+  
+  for (let i = 0; i < 3; i++) {
+    bit2 += String(Math.floor(Math.random() * 9));
+  }
+  
+  for (let i = 0; i < 4; i++) {
+    bit3 += String(Math.floor(Math.random() * 9));
+  }
+
+  if (type <= 1) {
+    // 1 XXX-XXX-XXXX
+    result = `1 ${bit1}-${bit2}-${bit3}`
+    
+  } else if (type <= 2) {
+    // 1 (XXX)XXX-XXXX
+    result = `1 (${bit1})${bit2}-${bit3}`;
+    
+  } else if (type <= 3) {
+    // 1(XXX)XXX-XXXX
+    result = `1(${bit1})${bit2}-${bit3}`;
+    
+  } else if (type <= 4) {
+    // 1 XXX XXX XXXX
+    result = `1 ${bit1} ${bit2} ${bit3}`
+    
+  } else if (type <= 5) {
+    // XXXXXXXXXX
+    result = `${bit1}${bit2}${bit3}`
+    
+  } else if (type <= 6) {
+    // XXX-XXX-XXXX
+    result = `${bit1}-${bit2}-${bit3}`
+    
+  } else {
+    //(555)555-5555
+    result = `(${bit1})${bit2}-${bit3}`
+    
+  }
```

Comment:
There's array hidden in here. The loop outside of the function iterating over the type does the same as loop iterating over array, while the latter being more clear.